Core/src/RequestTrait.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core;
+
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Utils;
+
+/**
+ * Provides a light wrapper around often used request modification functions.
+ */
+trait RequestTrait
+{
+    /**
+     * Adds/Modifys header values properly, i.e. replaces the each value's
+     * name with new value if it already exist, else adds the value to header
+     * line.
+     *
+     * @param Request $request The request object to modify.
+     * @param string $headerKey The header key to append upon.
+     * @param array $values The values to append to the header key.
+     * @param string $delimiter Delimiter to separate the each value of $values
+     * @return Request
+     */
+    private function appendOrModifyHeaders(
```

Comment:
The RequestWrapper has a function called `applyHeaders`. Does it make sense to put this in the wrapper too?

I understand we need this in the individual products too, but could we explore the possibility of using this as a static method?

Reply:
I noticed upon testing that just before invoking `ExponentialBackoff::execute()` method in `RequestWrappers::send()`, the request wrapper replaced the required header key with some request info headers in the [`applyHeaders()`](https://github.com/googleapis/google-cloud-php/blob/b964225a9b252f36f00bfcd6d82d0ad62bf2d356/Core/src/RequestWrapper.php#L217) method, thus for appending our required headers, we need to operate after all default headers are placed in the RequestWrapper layer.

So changed the logic to placing the initial header application in the `Exponential Backoff > execute()` method.